doc/install.rst

```diff
@@ -1,58 +1,655 @@
 
-Before starting
-===============
+Requirements
+============
+
+Minimum Requirements
+--------------------
+
+The non distributed (i.e. no MPI) version of Arbor can be compiled on Linux or OS X systems
+with very few tools.
+
+.. table:: Required Tools
+
+    =========== ============================================
+    Tool        Notes
+    =========== ============================================
+    git         To check out the code, min version 2.0.
+    cmake       To set up the build, min version 3.0.
+    compiler    A C++11 compiler. See `compilers <compilers_>`_.
+    =========== ============================================
+
+
+.. _compilers:
+
+Compilers
+~~~~~~~~~
+
+Arbor requires a C++ compiler that fully supports C++11 (we have plans to move
+to C++14 soon).
+We recommend using GCC or Clang, for which Arbor has been tested and optimised.
+
+.. table:: Supported Compilers
+
+    =========== ============ ============================================
+    Compiler    Min version  Notes
+    =========== ============ ============================================
+    GCC         5.2.0        5.1 probably works, 5.0 doesn't.
+    Clang       4.0          Clang 3.8 and later probably work.
+    Apple Clang 9
+    Intel       17.0.1       Needs GCC 5 or later for standard library.
+    =========== ============ ============================================
+
+.. Note::
+    The ``CC`` and ``CXX`` environment variables are used to specify the compiler executable
+    to the CMake build scripts. If these are not set, CMake will automatically choose a compiler.
+    On the system that the test below was performed, if the ``CC`` and ``CXX`` variables are
+    not set, CMake used ``/usr/bin/c++``, which was GCC version 4.8.5.
 
-We will require:
+    .. code-block:: bash
 
-  * git
-  * cmake 3.0
-  * C++11 compliant compiler
+        # check which version of GCC is available
+        $ g++ --version
+        g++ (GCC) 5.2.0
+        Copyright (C) 2015 Free Software Foundation, Inc.
 
-For GPU support:
+        # set environment variables for compilers
+        $ export CC=`which gcc`; export CXX=`which g++`;
 
-  * NVIDIA CUDA toolkit 8.0
+        # launch CMake
+        # the compiler version and path is given in the CMake output
+        $ cmake ..
+        -- The C compiler identification is GNU 5.2.0
+        -- The CXX compiler identification is GNU 5.2.0
+        -- Check for working C compiler: /cm/local/apps/gcc/5.2.0/bin/gcc
+        -- Check for working C compiler: /cm/local/apps/gcc/5.2.0/bin/gcc -- works
+        ...
 
-To make these docs you also need:
+.. Note::
+    Is is commonly assumed that to get the best performance one should use a vendor-specific
+    compiler (e.g. the Intel, Cray or IBM compilers). These compilers are often better at
+    auto-vectorizing loops, however for everything else GCC and Clang nearly always generate
+    more efficient code.
 
-  * Sphinx
+    The main computational loops in Arbor are based on
```

Comment:
I would say 'generated from' rather than 'based on' here, and say below that this code is vectorized explicitly.